### ISSUE TITLE
Rethink how and when the shop (and other) pages are getting created

### ISSUE THREAD

**Issue by peterfabian** [woocommerce/woocommerce#28318]
**Is your feature request related to a problem? Please describe.**
It's confusing for many users, e.g. if they once installed WC in the past, now install it again, but the pages won't get created, etc. 

**Describe the solution you'd like**
Maybe run it automatically on activation if the pages are missing, or add an inbox notification with a link to create them? 

**Describe alternatives you've considered**
Have confused users? 

**Additional context**
pbg9X-feq-p2

**Comment by pmcpinto** [comment]
I think that automatically creating the Woo pages in any case probably is the easiest solution. @elizaan36 what do you think?

**Comment by lkraav** [comment]
> I think that automatically creating the Woo pages in any case probably is the easiest solution. @elizaan36 what do you think?

I would definitely avoid auto-creating due to risk of spam, for example in development sites where de-/reactivation cycles can be multiples.

Just re-triggering "Create pages?" opt-in admin notification is the sane move, methinks.

**Comment by beaulebens** [comment]
A contextually-relevant point to prompt users for this might be when they go to edit products, or do other things specifically related to their store. We could then say something along the lines of "Looks like you're working with products, but your store isn't set up properly yet, would you like to [create the pages required for your store]?"

**Comment by elizaan36** [comment]
I agree that this is confusing and automatically creating the pages will make it easier to achieve store setup.

The settings to define these pages are hidden in *Settings > Advanced* with a vague description, "These pages need to be set so that WooCommerce knows where to send users to checkout." It's hard to understand so I can imagine a new user being stumped by this settings area.
<img width="1034" alt="Screen Shot 2020-11-18 at 9 32 22 PM" src="https://user-images.githubusercontent.com/5121465/99617967-0d763680-29e6-11eb-9322-c34dd0ec42b7.png">

Then the Tool to automatically create and define these pages is separately hidden in another area, *Utilities > Tools*. The action to "Create pages" is listed halfway down the long page with a vague description as well. "Create default WooCommerce pages. Note: This tool will install all the missing WooCommerce pages. Pages already defined and set up will not be replaced." There's little chance a new user would find this and if they did, they'd wonder what pages are being created.
<img width="993" alt="Screen Shot 2020-11-18 at 9 33 07 PM" src="https://user-images.githubusercontent.com/5121465/99618334-c3418500-29e6-11eb-99fa-78fc711d2448.png">

> I would definitely avoid auto-creating due to risk of spam, for example in development sites where de-/reactivation cycles can be multiples.

@lkraav Who would be on the receiving end of the spam, what amount of spam could be possible, and are there other solutions that could prevent that? 

> "Looks like you're working with products, but your store isn't set up properly yet, would you like to [create the pages required for your store]?"

@beaulebens I'm concerned that people might jump ship before they get to that point by not knowing how to create this functionality for their site, as a notice or other message can easily be missed or ignored.

> I think that automatically creating the Woo pages in any case probably is the easiest solution. @elizaan36 what do you think?

I agree that automatically creating all necessary pages makes the most sense. It seems like a must-have for basic usability and unless I'm missing something I don't see the benefit to users of not automatically setting up and defining default pages.

**Comment by lkraav** [comment]
> > I would definitely avoid auto-creating due to risk of spam, for example in development sites where de-/reactivation cycles can be multiples.
> 
> @lkraav Who would be on the receiving end of the spam, what amount of spam could be possible, and are there other solutions that could prevent that?

"Spam" here means auto-created pages that user doesn't want and probably ends up deleting every time they re-appear.

For example, we don't use WC core Shop page functionality at all - we build our own custom catalog.

To avoid any confusion, we delete all unused pages. WC pages magically reappearing with plugin activation cycles would be an annoying change.

**Comment by elizaan36** [comment]
Ah, I understand what you mean by spam now. I can see why that would be annoying.

I suppose the best solution would be to detect whether pages have already been defined for a site (in the case of de/reactivation cycles) and **don't** automatically create them for those stores.

As an alternative or fallback if that's not possible, we could continue presenting a notice. Pretty sure people are missing this since HE's in Quick start sessions had to help people find it, so maybe identifying contexts where it would be more relevant would be better as @beaulebens suggested.

Overall though, this is a technical task that we shouldn't expect people to have to do on their own so best if we could help new users out with as much as possible.

**Comment by pmcpinto** [comment]
Just adding a suggestion made by @rrennick in a P2 that could work

> It might be worth looking at storing timestamps or similar to determine whether WC had been active long enough for the site owner to be somewhat familiar with WC. And, if not, fire up the onboarding.

**Comment by Konamiman** [comment]
Hi @beaulebens, this one is quite old, do you think it's still relevant? If not we'll close it as part of our issues list maintenance.

**Comment by beaulebens** [comment]
As far as I'm aware, this is still relevant, but is somewhat of an edge case.

I'd propose that Ghidorah take ownership over this as part of general onboarding/getting up and running, but with lower priority. 

Definitely agree with Elizabeth above, that if pages have been previously created and deleted, we should not then automatically re-create them ever. I suspect we should opt for something lightweight, like a small notice/inbox message/something unobtrusive and which doesn't unintentionally create pages (requires interaction from the user). 

This is unlikely to be a problem for our target customer, who is assisted by a builder of some kind (they'll take care of this).

**Comment by chihsuan** [comment]
> if they once installed WC in the past, now install it again, but the pages won't get created, etc.


> Definitely agree with Elizabeth above, that if pages have been previously created and deleted, we should not then automatically re-create them ever. I suspect we should opt for something lightweight, like a small notice/inbox message/something unobtrusive and which doesn't unintentionally create pages (requires interaction from the user).

Given that we have a coming soon mode and launch your store task, I think we could show a notice or a task to notify the user about the missing pages that they can create before launching the store. @adrianduffell @verofasulo What do you think?

**Comment by adrianduffell** [comment]
> Given that we have a coming soon mode and launch your store task, I think we could show a notice or a task to notify the user about the missing pages that they can create before launching the store. @adrianduffell @verofasulo What do you think?

I think it is best to surface this in the launch your store task. It is not necessarily wrong to have missing pages -- they are optional fields -- and builder's custom functionality may break if pages are re-added. We could have higher confident in giving advice to fix missing pages in the LYS task, since it is a feature for merchant-led stores and they are more likely to need default settings.

Another idea is to re-run the core profiler when WooCommerce is installed again (i.e. when it was previously removed in the past, and now installed again, as opposed to performing an update). There could be a welcome screen shown that re-sets up the defaults including pages.

<img width="1812" alt="Image" src="https://github.com/user-attachments/assets/fca1fe10-65a9-4853-b1ca-bc0d86cb5203" />
